### ISSUE TITLE
Cannot relocate.replace flat ride after removing it in construction window (patch 3.5)

### ISSUE THREAD

**Issue by andy55699** [OpenRCT2/OpenRCT2#15887]
### Operating System

Windows 10,64-bit

### OpenRCT2 build

OpenRCT2, v0.3.5 (b9bc8d0) provided by GitHub

### Describe the issue

I have installed the latest patch 3.5 but somwhow I still encounter this bug.

Fix: [#15197] Cannot place flat ride after removing it in construction window.

 I encounter a bug when rebuilding/relocateing a flat ride that it cannot be demolished in the first step and I can only choose a new location to build the flat ride. However the original flat ride stays. Therefore 2 identical flat rides are built. When I demolish the new flat ride, the old one is also demolished.

 

This bug is similar to the shop relocation bug before which is fixed.

https://forums.openrct2.org/topic/5695-bug-cannot-relocate-a-shop-properly/?tab=comments#comment-26392

 

### Area(s) with issue?

Building the game

### Steps to reproduce

1. Relocate the flat ride.

### Attachments

![1](https://user-images.githubusercontent.com/92691705/141135945-d4f28852-7ce3-4daf-905d-26ebf4927564.png)
![2](https://user-images.githubusercontent.com/92691705/141135953-7a874c64-8f8b-419b-93e7-b1e19446da8c.png)
![3](https://user-images.githubusercontent.com/92691705/141135958-055546df-65fb-44fc-9748-2b62b7899f7a.png)

Saved game in ZIP
[Bug saved game.zip](https://github.com/OpenRCT2/OpenRCT2/files/7513622/Bug.saved.game.zip)

**Comment by ZehMatt** [comment]
I've tried this on a new save and everything works as expected. I tried your save and it fails to demolish the ride properly as it claims the track element (flat ride element) can not be found, did you mess around with the tile inspector by any chance?

**Comment by andy55699** [comment]
> I've tried this on a new save and everything works as expected. I tried your save and it fails to demolish the ride properly as it claims the track element (flat ride element) can not be found, did you mess around with the tile inspector by any chance?

[Ayers Adventure bug.zip](https://github.com/OpenRCT2/OpenRCT2/files/7529517/Ayers.Adventure.bug.zip)
I have saved a version of a completely new and untouched park without any edits and cheats. You would see the bug still exists in this save.

**Comment by LostInRiverview** [comment]
I've encountered the same bug. In my case, I was in Bumbly Beach (my game is linked to an RCT1 installation) and tried to relocate the pre-built Ferris Wheel. When trying to delete the Wheel in the construction window (in order to move it, rather than demolishing the ride entirely) I got a "can't remove this..." error and the ride was duplicated. I was able to place multiple ferris wheels that were all basically treated as separate stations on the same ride; when each wheel was given an exit and entrance, setting the ride to test mode would start all four wheels moving simultaneously. This behavior persists after reloading OpenRCT2, but I haven't encountered it in other RCT1 parks so far. I'm running the same version as andy55699: v0.3.5 (b9bc8d0)
